runbotics/runbotics-desktop/src/core/bpm/desktop-runner/desktop-runner.service.ts

```diff
@@ -240,4 +239,26 @@ export class DesktopRunnerService implements OnModuleInit {
         this.logger.error(notFoundErrorMessage);
         throw new Error(notFoundErrorMessage);
     }
-}
+
+    private checkIfFileSystemEntryPresentInDirectory(directoryPath: string, directoryName: string, fileSystemEntires: string[]): boolean {
+        const isPresent = fileSystemEntires.every(fileSystemEntry => {
+            const path = directoryPath + '\\' + directoryName + '\\' + fileSystemEntry;
+            return existsSync(path);
+        });
+
+        return isPresent;
+    }
+
+    private detectExtensionsDirectories(dirent: Dirent) {
+        return (
+            dirent.isDirectory() &&
+            !dirent.name.startsWith('.') &&
+            this.checkIfFileSystemEntryPresentInDirectory(dirent.path, dirent.name, [
+                'dist',
+                'node_modules',
+                'package.json',
+                'package-lock.json'
```

Comment:
there are 3 options each for every package manager `package-lock.json` for npm, `yarn.lock` for yarn and `pnpm-lock.yaml` for pnpm